### PR TITLE
[Done] Translation of faq-functions.md ..

### PR DESCRIPTION
**Pull request by 3imed-jaberi**

### CODE REVIEW

File: content/docs/faq-functions.md [hunk 1]
```diff
@@ -178,9 +178,9 @@ class Alphabet extends React.Component {
 }
 ```
 
-#### Example: Passing params using data-attributes {#example-passing-params-using-data-attributes}
+#### مثال: تمرير المُعامِلات باستخدام خاصيّات البيانات {#example-passing-params-using-data-attributes}
```
Comment (Fcmam5) [inline content/docs/faq-functions.md]: Remove the lines above (from 147 to here) :smile:

File: content/docs/faq-functions.md [hunk 2]
```diff
@@ -302,13 +302,13 @@ class Searchbox extends React.Component {
...
+إنّ [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) هو عبارة عن طريقة لوضع الدالة في طابور لتنفيذه في المتصفح في الوقت المثالي لتحسين أداء التصيير. تُطلَق الدالة التي توضع في الطابور باستخدام `requestAnimationFrame` في الإطار الزمني التالي. سيعمل المتصفح بجد لضمان الحصول على 60 إطار في الثانية (60 fps). إن لم يكن المتصفح قادرًا على ذلك فسـ*يحد* عدد الإطارات في الثانية. على سبيل المثال قد يكون جهازك قادر على التعامل فقط مع 30 إطار بالثانية لذا ستحصل على 30 إطار بالثانية فقط. إنّ استخدام `requestAnimationFrame` للخنق هو تقنية مفيدة تمنع من إجراء أكثر من 60 تحديث في الثانية. إن كنت تجري 100 تحديث في الثانية فسيؤدي ذلك إلى إنشاء عمل إضافي على المتصفح والذي لن يراه المستخدم على أية حال.
 
->**Note:**
+>**ملاحظة:**
 >
->Using this technique will only capture the last published value in a frame. You can see an example of how this optimization works on [`MDN`](https://developer.mozilla.org/en-US/docs/Web/Events/scroll)
+>استخدام هذه التقنية سيلتقط فقط آخر قيمة منشورة في الإطار. بإمكانك رؤية مثال حول كيفية عمل هذا الضبط من [هنا](https://developer.mozilla.org/en-US/docs/Web/Events/scroll).
 
 ```jsx
```
Comment (Fcmam5) [inline content/docs/faq-functions.md]: This code example is not translated

File: content/docs/faq-functions.md [hunk 3]
```diff
@@ -349,6 +349,6 @@ class ScrollListener extends React.Component {
 }
 ```
 
-#### Testing your rate limiting {#testing-your-rate-limiting}
+#### اختبار حدود معدل تحديث الإطار لديك {#testing-your-rate-limiting}
 
-When testing your rate limiting code works correctly it is helpful to have the ability to fast forward time. If you are using [`jest`](https://facebook.github.io/jest/) then you can use [`mock timers`](https://facebook.github.io/jest/docs/en/timer-mocks.html) to fast forward time. If you are using `requestAnimationFrame` throttling then you may find [`raf-stub`](https://github.com/alexreardon/raf-stub) to be a useful tool to control the ticking of animation frames.
+عند اختبار حدود معدل تحديث الإطار من المفيد امتلاك القدرة على تمرير الزمن بسرعة. إن كنت تستخدم [`jest`](https://facebook.github.io/jest/) بإمكانك استخدام [`محاكيات الوقت`](https://facebook.github.io/jest/docs/en/timer-mocks.html) لتمرير الوقت بسرعة. إن كنت تستخدم الخنق عن طريق `requestAnimationFrame` فهنالك الأداة [`raf-stub`](https://github.com/alexreardon/raf-stub) مفيدة للتحكم بضبط تحريك الإطارات.
```
Comment (Fcmam5) [inline content/docs/faq-functions.md]:
```suggestion
عند اختبار حدود معدل تحديث الإطار من المفيد امتلاك القدرة على تمرير الزمن بسرعة. إن كنت تستخدم [`jest`](https://facebook.github.io/jest/) بإمكانك استخدام [`محاكيات المؤقّتات (mock timers)`](https://facebook.github.io/jest/docs/en/timer-mocks.html) لتمرير الوقت بسرعة. إن كنت تستخدم الخنق عن طريق `requestAnimationFrame` فهنالك الأداة [`raf-stub`](https://github.com/alexreardon/raf-stub) مفيدة للتحكم بضبط تحريك الإطارات.
```

File: content/docs/faq-functions.md [hunk 4]
```diff
@@ -29,71 +29,71 @@ class Foo extends Component {
...
+بشكلٍ عام نعم، لا مشكلة في ذلك، وهي عادةً الطريقة الأسهل لتمرير المُعامِلات إلى دوال ردود النداء.
 
-If you do have performance issues, by all means, optimize!
+إن كانت لديك مشاكل في الأداء، فحاول تحسينه عن طريق الطرق المشروحة في [المستندات](/docs/getting-started.html).
 
-### Why is binding necessary at all? {#why-is-binding-necessary-at-all}
+### لماذا من الضروري إجراء الربط أساسًا؟ {#why-is-binding-necessary-at-all}
 
-In JavaScript, these two code snippets are **not** equivalent:
+في JavaScript **لا** تكون الشيفرتان التاليتان متساويتين:
```
[the remark this answers is not in the document]
Comment (3imed-jaberi) [reply content/docs/faq-functions.md]: I think there is no need to put the 2 words bold .. just the negation word  ...

[the remark this answers is not in the document]
Comment (Fcmam5) [reply content/docs/faq-functions.md]: The negation should be highlighted, so I believe that the whole negation expression should be put in bold. But it's okay.

**Pull request merged**